### PR TITLE
test(pkg/quotes): migrate test-infra to testify for pkg/quotes pkg

### LINKED ISSUE #2928: migrate test-infra to testify for `pkg/quotes` pkg

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve? 
migrate test-infra to testify for pkg/quotes pkg, close  #2928

### What is changed and how it works?

1. migrate test-infra from gocheck to testify
2. migrate leak test from leaktest to goleak

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: pkg/quotes/quotes_test.go [hunk 1]
```diff
@@ -38,12 +32,13 @@ func (s *quotesSuite) TestQuoteSchema(c *check.C) {
 	}
 	for _, testCase := range cases {
 		name := QuoteSchema(testCase.schema, testCase.table)
-		c.Assert(name, check.Equals, testCase.expected)
+		require.Equal(t, name, testCase.expected)
```
Comment (amyangfei) [inline pkg/quotes/quotes_test.go:35]: I find the function is [func Equal(t TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{})](https://pkg.go.dev/github.com/stretchr/testify/require#Equal) actually, better to use `require.Equal(t, testCase.expected, name)
Ditto for other places we use `require.Equal`

**Review by ben1009 — APPROVED** [review]

File: pkg/quotes/quotes_test.go (hunk 1 above)
Comment (amyangfei) [inline pkg/quotes/quotes_test.go:35]:
```suggestion
		require.Equal(t, testCase.expected, name)
```

File: pkg/quotes/quotes_test.go [hunk 2]
```diff
@@ -55,12 +50,13 @@ func (s *quotesSuite) TestQuoteName(c *check.C) {
 	}
 	for _, testCase := range cases {
 		escaped := QuoteName(testCase.name)
-		c.Assert(escaped, check.Equals, testCase.expected)
+		require.Equal(t, escaped, testCase.expected)
```
Comment (amyangfei) [inline pkg/quotes/quotes_test.go:53]:
```suggestion
		require.Equal(t, testCase.expected, escaped)
```

File: pkg/quotes/quotes_test.go [hunk 3]
```diff
@@ -73,6 +69,6 @@ func (s *quotesSuite) TestEscapeName(c *check.C) {
 	}
 	for _, testCase := range cases {
 		escaped := EscapeName(testCase.name)
-		c.Assert(escaped, check.Equals, testCase.expected)
+		require.Equal(t, escaped, testCase.expected)
```
Comment (amyangfei) [inline pkg/quotes/quotes_test.go:72]:
```suggestion
		require.Equal(t, testCase.expected, escaped)
```

**Review by amyangfei — APPROVED** [review]

**Pull request merged**